src/Chapter2.hs

```diff
@@ -841,7 +851,13 @@ list.
 
 🕯 HINT: Use the 'cycle' function
 -}
-rotate = error "rotate: Not implemented!"
+rotate :: Int -> [Int] -> [Int]
+rotate n l
+  | n < 0 = []
+  | null l = []
+  | otherwise = 
+      let len = length l
+      in take len (drop (mod n len) (cycle l)) -- mod because a rotate of len just returns the same thing
```

Comment:
Really nice catch, it makes the implementation way more efficient! 👍🏼 

You can also add another trick to speed up – return the list itself if `mod` is 0 😉